src/Symfony/Component/Console/Command/Command.php

```diff
@@ -241,6 +241,10 @@ public function run(InputInterface $input, OutputInterface $output)
             $this->interact($input, $output);
         }
 
+        if ($input->hasArgument('command') && null === $input->getArgument('command')) {
+            $input->setArgument('command', $this->getName());
```

Comment:
Great suggestion. I'll add:

```php
        // The command name argument is often omitted when a command is executed directly with its run() method.
        // It will fail the validation since the command argument is required by the application.
```